source/processes/fs.cpp

```diff
@@ -2112,24 +2116,62 @@ decltype(HLE::OS::ServiceHelper::SendReply) FakeFS::UserCommandHandler(FakeThrea
         break;
     }
 
-    case 0x83a: // GetSpecialContentIndex
+    case GetSpecialContentIndex::id:
     {
         auto media_type = MediaType { static_cast<uint8_t>(thread.ReadTLS(0x84)) };
-        auto content_type = thread.ReadTLS(0x90) & 0xff;
-        if (/*media_type == MediaType::GameCard*/ true) {
-            thread.WriteTLS(0x80, IPC::CommandHeader::Make(0, 2, 0).raw);
-            thread.WriteTLS(0x84, RESULT_OK);
-            if (content_type == 1) {
-                thread.WriteTLS(0x88, Meta::to_underlying(Loader::NCSDPartitionId::UpdateData));
-            } else if (content_type == 2) {
-                thread.WriteTLS(0x88, Meta::to_underlying(Loader::NCSDPartitionId::Manual));
-            } else if (content_type == 3) {
-                thread.WriteTLS(0x88, Meta::to_underlying(Loader::NCSDPartitionId::DownloadPlayChild));
+        auto content_type = SpecialContentType { static_cast<SpecialContentType>(thread.ReadTLS(0x90) & 0xff) };
+        if (media_type == MediaType::GameCard) {
+            Loader::NCSDPartitionId target_ncsd_id;
+            switch (content_type) {
+                case Platform::FS::SpecialContentType::UpdateData:
+                    target_ncsd_id = Loader::NCSDPartitionId::UpdateData;
+                    break;
+                case Platform::FS::SpecialContentType::Manual:
+                    target_ncsd_id = Loader::NCSDPartitionId::Manual;
+                    break;
+                case Platform::FS::SpecialContentType::DownloadPlayChild:
+                    target_ncsd_id = Loader::NCSDPartitionId::DownloadPlayChild;
+                    break;
+                case Platform::FS::SpecialContentType::UpdateDataNew3DS:
+                    target_ncsd_id = Loader::NCSDPartitionId::UpdateDataNew3DS;
+                    break;
+                default:
+                    throw Mikage::Exceptions::Invalid("GetSpecialContentType: Invalid special content type {:#x}", Meta::to_underlying(content_type));
+            }
+            
+            auto gamecard = thread.GetOS().setup.gamecard.get();
+            assert(gamecard);
```

Comment:
`assert` is the wrong tool here, since an emulated application could (theoretically) call GetSpecialContentType without a card inserted.

So this should either warrant a runtime-check with `Mikage::Exceptions::Invalid`, or it should be dropped in favor of just crashing with a nullptr access.